heap/max_heap.py

```diff
@@ -21,15 +21,18 @@ def insert(self, value):
         this method needs to ensure that the heap property is maintained after 
         the top most element has been removed. """
     def delete(self):
+        self._bubble_up(len(self.storage) - 1)
```

Comment:
you don't need this here. your heap _should_ already be in the right position, so _bubble_up() in a tree that's ordered will do nothing